### PR TITLE
[Data-565] Fix for flaky join pointcloud naive test

### PR DESCRIPTION
**Pull request by bhaney**
The bug was an edge-case when reading the batches of points from a channel into the final point cloud would not read all the points in the channel. 

The loop would continue reading data into the final point cloud if `dataLastTime || atomic.LoadInt32(&activeReaders) > 0`

If there was no data in the channel, then the loop would wait 5 ms and then would set `dataLastTime = false`.

There was a situation where `dataListTime` was false, but `atomic.LoadInt32(&activeReaders)` had become 0 only within those 5 ms when `dataLastTime` was being set to false, meaning there was still some last minute data in the channel. The solution is to do one last read to make sure all of the data is out of the pipeline.

### CODE REVIEW

**Comment by bhaney** [comment]
lol failed the test again

File: pointcloud/merging.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+	for i := 0; i < lastBatches; i++ {
+		lastPoints := <-finalPoints
+		logger.Debugf("number of points in batch %d: %d", i, len(lastPoints))
+		for _, p := range lastPoints {
+			myErr := pcTo.Set(p.P, p.D)
+			if myErr != nil {
+				err = myErr
+			}
+		}
+	}
```
Comment (bhaney) [inline pointcloud/merging.go:116]: This is the fix, esssentially

File: pointcloud/merging.go [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+				loopCopy := loop
+				utils.PanicCapturingGo(func() { f(loopCopy) })
+			}
+			wg.Wait()
+		})
+	}
+	var pcTo PointCloud
+	var err error
+
+	dataLastTime := false // if there was data in the channel in the previous loop, continue reading.
```
Comment (edaniels) [inline pointcloud/merging.go:78]: this may be a naive question and can be discussed later, but how fast is this code compared to a simpler version. This feels very complicated.

File: pointcloud/merging.go [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+	for dataLastTime || atomic.LoadInt32(&activeReaders) > 0 {
```
Comment (edaniels) [inline pointcloud/merging.go:79]: why is this looping like this versus waiting on a channel and/or waitgroup?

File: pointcloud/merging.go [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+		select {
+		case ps := <-finalPoints:
+			for _, p := range ps {
+				if pcTo == nil {
+					if p.D == nil {
+						pcTo = NewAppendOnlyOnlyPointsPointCloud(len(cloudFuncs) * 640 * 800)
```
Comment (edaniels) [inline pointcloud/merging.go:85]: what's the 640*800 for? should it be a const?

**Review by edaniels — APPROVED** [review]
I'm okay with this fixing the flake but if this code is used a lot, I'd love if we can benchmark it and possibly simplify it greatly.

File: pointcloud/merging.go (hunk 2 above)
Comment (bhaney) [reply pointcloud/merging.go:78]: I will make a ticket for simplifying this merging function.
